File: src/baekjoon/_4000/_4673_implement/Main.java
package baekjoon._4000._4673_implement;

public class Main {
    private static final int SIZE = 10000;

    public static void main(String[] args) {
        StringBuilder answer = new StringBuilder();

        boolean[] selfNumbers = new boolean[SIZE + 1];

        for (int i = 1; i <= SIZE; i++) {
            int sum = i;
            String str = String.valueOf(i);
            for (int index = 0; index < str.length(); index++) {
                sum += str.charAt(index) - '0';
            }
            if (sum > SIZE) continue;
            selfNumbers[sum] = true;
        }

        for (int i = 1; i <= SIZE; i++) {
            if (!selfNumbers[i]) {
                answer.append(i);
                answer.append("\n");
            }
        }

        System.out.println(answer);
    }
}

File: src/baekjoon/_4000/_4673_implement/main.kt
//val SIZE = 10001
//fun main() {
//       val selfNumbers = Array(SIZE) { false }
//
//       var result = StringBuilder()
//       for(number in 1 until SIZE) {
//              selfNumbers[makeSelfNumber(number)] = true
//       }
//
//       for(index in 1 until SIZE) {
//              if (!selfNumbers[index]) {
//                     result.append(index).append("\n")
//              }
//       }
//       println(result.toString())
//}
//
//
//fun makeSelfNumber(number: Int): Int {
//       val str = number.toString()
//       var sum = number
//       for(ch in str) {
//           sum += ch.digitToInt()
//       }
//       if(sum >= SIZE) sum = 0;
//       return sum
//}
